{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "43ca2901",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "c3cb45da",
   "metadata": {},
   "outputs": [],
   "source": [
    "path='D:/FILES/University of Toronto/Subjects/Fall 2022/NLP/project/PyOverflow/Data/Training/built-in-exceptions.json'\n",
    "df=pd.read_json(path_or_buf=path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b9f1a49e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove empty lists from subtopics location 0\n",
    "#df['Subtopics'].loc[0]=list(filter(None, df['Subtopics'].loc[0]))\n",
    "data=pd.DataFrame(df['Subtopics'].loc[0])\n",
    "data=data[['Topic','content']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "a01fbca4",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in data.index:\n",
    "    data['content'].loc[i]=' '.join(data['content'].loc[i])\n",
    "    data['content'].loc[i]=data['content'].loc[i].replace('\\\\u',' ')\n",
    "    data['content'].loc[i]=data['content'].loc[i].replace('\\n',' ')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "088b680c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Topic</th>\n",
       "      <th>content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Exception context</td>\n",
       "      <td>When raising a new exception while another exc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Inheriting from built-in exceptions</td>\n",
       "      <td>User code can create subclasses that inherit f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Base classes</td>\n",
       "      <td>The following exceptions are used mostly as ba...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Concrete exceptions</td>\n",
       "      <td>The following exceptions are the exceptions th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Warnings</td>\n",
       "      <td>The following exceptions are used as warning c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Exception groups</td>\n",
       "      <td>The following are used when it is necessary to...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Exception hierarchy</td>\n",
       "      <td>The class hierarchy for built-in exceptions is:</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Topic  \\\n",
       "0                    Exception context   \n",
       "1  Inheriting from built-in exceptions   \n",
       "2                         Base classes   \n",
       "3                  Concrete exceptions   \n",
       "4                             Warnings   \n",
       "5                     Exception groups   \n",
       "6                  Exception hierarchy   \n",
       "\n",
       "                                             content  \n",
       "0  When raising a new exception while another exc...  \n",
       "1  User code can create subclasses that inherit f...  \n",
       "2  The following exceptions are used mostly as ba...  \n",
       "3  The following exceptions are the exceptions th...  \n",
       "4  The following exceptions are used as warning c...  \n",
       "5  The following are used when it is necessary to...  \n",
       "6    The class hierarchy for built-in exceptions is:  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "787cd27f",
   "metadata": {},
   "outputs": [],
   "source": [
    "df=df[['Topic','content']]\n",
    "df['content']=' '.join(df['content'][0])\n",
    "df['content']=df['content'][0].replace('\\\\u',' ')\n",
    "df['content']=df['content'][0].replace('\\n',' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8aeb345d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Topic</th>\n",
       "      <th>content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Built-in Exceptions</td>\n",
       "      <td>In Python, all exceptions must be instances of...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Topic                                            content\n",
       "0  Built-in Exceptions  In Python, all exceptions must be instances of..."
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "3be59691",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in data.index:\n",
    "    df['content']=df['content'][0]+'. '+data['content'].loc[i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "4e15d470",
   "metadata": {},
   "outputs": [],
   "source": [
    "#split the df[content] into sentences\n",
    "df['content'][0]=df['content'][0].split('.')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "fc61bdc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['In Python, all exceptions must be instances of a class that derives from BaseException',\n",
       " '  In a try statement with an except clause that mentions a particular class, that clause also handles any exception classes derived from that class (but not exception classes from which it is derived)',\n",
       " '  Two exception classes that are not related via subclassing are never equivalent, even if they have the same name',\n",
       " ' The built-in exceptions listed below can be generated by the interpreter or built-in functions',\n",
       " '  Except where mentioned, they have an “associated value” indicating the detailed cause of the error',\n",
       " '  This may be a string or a tuple of several items of information (e',\n",
       " 'g',\n",
       " ', an error code and a string explaining the code)',\n",
       " '  The associated value is usually passed as arguments to the exception class’s constructor',\n",
       " ' User code can raise built-in exceptions',\n",
       " '  This can be used to test an exception handler or to report an error condition “just like” the situation in which the interpreter raises the same exception; but beware that there is nothing to prevent user code from raising an inappropriate error',\n",
       " ' The built-in exception classes can be subclassed to define new exceptions; programmers are encouraged to derive new exceptions from the Exception class or one of its subclasses, and not from BaseException',\n",
       " '  More information on defining exceptions is available in the Python Tutorial under User-defined Exceptions',\n",
       " '',\n",
       " ' When raising a new exception while another exception is already being handled, the new exception’s __context__ attribute is automatically set to the handled exception',\n",
       " '  An exception may be handled when an except or finally clause, or a with statement, is used',\n",
       " ' This implicit exception context can be supplemented with an explicit cause by using from with raise: The expression following from must be an exception or None',\n",
       " ' It will be set as __cause__ on the raised exception',\n",
       " ' Setting __cause__ also implicitly sets the __suppress_context__ attribute to True, so that using raise new_exc from None effectively replaces the old exception with the new one for display purposes (e',\n",
       " 'g',\n",
       " ' converting KeyError to AttributeError), while leaving the old exception available in __context__ for introspection when debugging',\n",
       " ' The default traceback display code shows these chained exceptions in addition to the traceback for the exception itself',\n",
       " ' An explicitly chained exception in __cause__ is always shown when present',\n",
       " ' An implicitly chained exception in __context__ is shown only if __cause__ is None and __suppress_context__ is false',\n",
       " ' In either case, the exception itself is always shown after any chained exceptions so that the final line of the traceback always shows the last exception that was raised',\n",
       " '',\n",
       " ' User code can create subclasses that inherit from an exception type',\n",
       " ' It’s recommended to only subclass one exception type at a time to avoid any possible conflicts between how the bases handle the args attribute, as well as due to possible memory layout incompatibilities',\n",
       " '',\n",
       " ' The following exceptions are used mostly as base classes for other exceptions',\n",
       " '',\n",
       " ' The following exceptions are the exceptions that are usually raised',\n",
       " ' The following exceptions are kept for compatibility with previous versions; starting from Python 3',\n",
       " '3, they are aliases of OSError',\n",
       " '',\n",
       " ' The following exceptions are used as warning categories; see the Warning Categories documentation for more details',\n",
       " '',\n",
       " ' The following are used when it is necessary to raise multiple unrelated exceptions',\n",
       " ' They are part of the exception hierarchy so they can be handled with except like all other exceptions',\n",
       " ' In addition, they are recognised by except*, which matches their subgroups based on the types of the contained exceptions',\n",
       " '',\n",
       " ' The class hierarchy for built-in exceptions is:']"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['content'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f9375d58",
   "metadata": {},
   "outputs": [],
   "source": [
    "#make a new df with the sentences\n",
    "df2=pd.DataFrame(df['content'][0],columns=['content'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "7ef27ecd",
   "metadata": {},
   "outputs": [],
   "source": [
    "df2['topic']=df['Topic'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "21e281a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>content</th>\n",
       "      <th>topic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>In Python, all exceptions must be instances of...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>In a try statement with an except clause tha...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Two exception classes that are not related v...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>The built-in exceptions listed below can be g...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Except where mentioned, they have an “associ...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>This may be a string or a tuple of several i...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>, an error code and a string explaining the code)</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>The associated value is usually passed as ar...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>User code can raise built-in exceptions</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>This can be used to test an exception handle...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>The built-in exception classes can be subclas...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>More information on defining exceptions is a...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td></td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>When raising a new exception while another ex...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>An exception may be handled when an except o...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>This implicit exception context can be supple...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>It will be set as __cause__ on the raised exc...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Setting __cause__ also implicitly sets the __...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>converting KeyError to AttributeError), while...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>The default traceback display code shows thes...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>An explicitly chained exception in __cause__ ...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>An implicitly chained exception in __context_...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>In either case, the exception itself is alway...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td></td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>User code can create subclasses that inherit ...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>It’s recommended to only subclass one excepti...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td></td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>The following exceptions are used mostly as b...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td></td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>The following exceptions are the exceptions t...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>The following exceptions are kept for compati...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>3, they are aliases of OSError</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td></td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>The following exceptions are used as warning ...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td></td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>The following are used when it is necessary t...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>They are part of the exception hierarchy so t...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>In addition, they are recognised by except*, ...</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td></td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>The class hierarchy for built-in exceptions is:</td>\n",
       "      <td>Built-in Exceptions</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              content                topic\n",
       "0   In Python, all exceptions must be instances of...  Built-in Exceptions\n",
       "1     In a try statement with an except clause tha...  Built-in Exceptions\n",
       "2     Two exception classes that are not related v...  Built-in Exceptions\n",
       "3    The built-in exceptions listed below can be g...  Built-in Exceptions\n",
       "4     Except where mentioned, they have an “associ...  Built-in Exceptions\n",
       "5     This may be a string or a tuple of several i...  Built-in Exceptions\n",
       "6                                                   g  Built-in Exceptions\n",
       "7   , an error code and a string explaining the code)  Built-in Exceptions\n",
       "8     The associated value is usually passed as ar...  Built-in Exceptions\n",
       "9             User code can raise built-in exceptions  Built-in Exceptions\n",
       "10    This can be used to test an exception handle...  Built-in Exceptions\n",
       "11   The built-in exception classes can be subclas...  Built-in Exceptions\n",
       "12    More information on defining exceptions is a...  Built-in Exceptions\n",
       "13                                                     Built-in Exceptions\n",
       "14   When raising a new exception while another ex...  Built-in Exceptions\n",
       "15    An exception may be handled when an except o...  Built-in Exceptions\n",
       "16   This implicit exception context can be supple...  Built-in Exceptions\n",
       "17   It will be set as __cause__ on the raised exc...  Built-in Exceptions\n",
       "18   Setting __cause__ also implicitly sets the __...  Built-in Exceptions\n",
       "19                                                  g  Built-in Exceptions\n",
       "20   converting KeyError to AttributeError), while...  Built-in Exceptions\n",
       "21   The default traceback display code shows thes...  Built-in Exceptions\n",
       "22   An explicitly chained exception in __cause__ ...  Built-in Exceptions\n",
       "23   An implicitly chained exception in __context_...  Built-in Exceptions\n",
       "24   In either case, the exception itself is alway...  Built-in Exceptions\n",
       "25                                                     Built-in Exceptions\n",
       "26   User code can create subclasses that inherit ...  Built-in Exceptions\n",
       "27   It’s recommended to only subclass one excepti...  Built-in Exceptions\n",
       "28                                                     Built-in Exceptions\n",
       "29   The following exceptions are used mostly as b...  Built-in Exceptions\n",
       "30                                                     Built-in Exceptions\n",
       "31   The following exceptions are the exceptions t...  Built-in Exceptions\n",
       "32   The following exceptions are kept for compati...  Built-in Exceptions\n",
       "33                     3, they are aliases of OSError  Built-in Exceptions\n",
       "34                                                     Built-in Exceptions\n",
       "35   The following exceptions are used as warning ...  Built-in Exceptions\n",
       "36                                                     Built-in Exceptions\n",
       "37   The following are used when it is necessary t...  Built-in Exceptions\n",
       "38   They are part of the exception hierarchy so t...  Built-in Exceptions\n",
       "39   In addition, they are recognised by except*, ...  Built-in Exceptions\n",
       "40                                                     Built-in Exceptions\n",
       "41    The class hierarchy for built-in exceptions is:  Built-in Exceptions"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "2d92d362",
   "metadata": {},
   "outputs": [],
   "source": [
    "df2.to_csv('D:/FILES/University of Toronto/Subjects/Fall 2022/NLP/project/PyOverflow/Data/Training/built-in-exceptions.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3c88e72",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7.13 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.13"
  },
  "vscode": {
   "interpreter": {
    "hash": "415686034d28ca83dfb397db2f59cd0b342825bd24c1e67598292e91597816cd"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
